light-prover-sdk/tests/genericProver.test.ts

```diff
@@ -0,0 +1,353 @@
+import { assert, expect } from "chai";
+import { Prover } from "../src/prover";
+import { Idl } from "@coral-xyz/anchor";
+
+let circomlibjs = require("circomlibjs");
+import { Keypair as SolanaKeypair } from "@solana/web3.js";
+import * as anchor from "@coral-xyz/anchor";
+
+import { it } from "mocha";
+const chai = require("chai");
+const chaiAsPromised = require("chai-as-promised");
+
+// Load chai-as-promised support
+chai.use(chaiAsPromised);
+import {
+  FEE_ASSET,
+  Provider as LightProvider,
+  MINT,
+  Transaction,
+  TransactionParameters,
+  Action,
+  Relayer,
+  Utxo,
+  Account,
+  MerkleTree,
+  IDL_VERIFIER_PROGRAM_ZERO,
+} from "../../light-zk.js/src";
+import { ProofInputs } from "../src/generics";
+
+process.env.ANCHOR_PROVIDER_URL = "http://127.0.0.1:8899";
+process.env.ANCHOR_WALLET = process.env.HOME + "/.config/solana/id.json";
+
+describe("Test Prover Functional", () => {
+  let seed32 = new Uint8Array(32).fill(1).toString();
+  let depositAmount = 20_000;
+  let depositFeeAmount = 10_000;
+
+  let mockPubkey = SolanaKeypair.generate().publicKey;
+  let mockPubkey2 = SolanaKeypair.generate().publicKey;
+  let mockPubkey3 = SolanaKeypair.generate().publicKey;
+  let poseidon,
+    lightProvider: LightProvider,
+    deposit_utxo1,
+    relayer,
+    keypair,
+    paramsDeposit,
+    paramsWithdrawal;
+
+  before(async () => {
+    poseidon = await circomlibjs.buildPoseidonOpt();
+    // TODO: make fee mandatory
+    relayer = new Relayer(
```

Comment:
why do you need a relayer in this test it should work without a blockchain then you don't need a relayer